core/invoker/src/main/scala/whisk/core/invoker/InvokerReactive.scala

```diff
@@ -113,13 +113,13 @@ class InvokerReactive(
   private val ack = (tid: TransactionId,
                      activationResult: WhiskActivation,
                      blockingInvoke: Boolean,
-                     controllerInstance: InstanceId,
+                     controllerInstance: StringInstanceId,
                      userId: UUID) => {
     implicit val transid: TransactionId = tid
 
     def send(res: Either[ActivationId, WhiskActivation], recovery: Boolean = false) = {
       val msg = CompletionMessage(transid, res, instance)
-      producer.send(s"completed${controllerInstance.toInt}", msg).andThen {
+      producer.send(topic = controllerInstance.topicName("completed"), msg).andThen {
```

Comment:
this is done on every active ack - which now requires a regex match --- we should avoid this (moving the match to the constructor would achieve that).

Reply:
in fact if we move the validation to the constructor, then the string interpolation can remain as is, with a change from toInt to asString.

Reply:
Good point - I'll move it.

Reply:
Based on problems moving validation to constructor, I can instead cache the regex+validation result instead?

Reply:
how strict do we want this validation to be? a `require` can be elided at compile time.